File: app/src/main/java/com/veganbeauty/app/features/community/com_feed/CommunityViewModelFactory.java
package com.veganbeauty.app.features.community.com_feed;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.veganbeauty.app.data.repository.CommunityRepository;

public class CommunityViewModelFactory implements ViewModelProvider.Factory {
    private final CommunityRepository repository;

    public CommunityViewModelFactory(CommunityRepository repository) {
        this.repository = repository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(CommunityViewModel.class)) {
            return (T) new CommunityViewModel(repository);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: app/src/main/java/com/veganbeauty/app/features/community/com_feed/CommunityViewModelFactory.kt
package com.veganbeauty.app.features.community.com_feed

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.veganbeauty.app.data.repository.CommunityRepository

class CommunityViewModelFactory(private val repository: CommunityRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(CommunityViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return CommunityViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}
